File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = ''
    for i in range(len(s)):
        if s[i] == ' ':
            answer += ' '
            continue
            
        if 65 <= ord(s[i]) <= 90:
            tmp = ord(s[i]) + n
            if tmp > 90:
                tmp -= 90
                tmp += 64
        elif 97 <= ord(s[i]) <= 122:
            tmp = ord(s[i]) + n
            if tmp > 122:
                tmp -= 122
                tmp += 96
        answer += chr(tmp)
    return answer

'''
A 65
Z 90
a 97
z 122
'''

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                answer += " ";
                continue;
            }
            
            int tmp = (int) s.charAt(i);
            if (tmp >= 65 && tmp <= 90) {
                tmp += n;
                if (tmp > 90) {
                    tmp -= 90;
                    tmp += 64;
                }
                
            } else if (tmp >= 97 && tmp <= 122) {
                tmp += n;
                if (tmp > 122) {
                    tmp -= 122;
                    tmp += 96;
                }
            }
            
            answer += (char) tmp;
        }
        return answer;
    }
}